application/src/js/charts/rd-data-tools.js

```diff
@@ -83,12 +82,10 @@ function formatNumberWithDecimalPlaces(value, dp) {
             number = number.replace("%", "");
         } finally {
             var formatted = (number * 1).toLocaleString("en-uk", { minimumFractionDigits: dp, maximumFractionDigits: dp });
-            if (formatted !== "NaN") {
-                return formatted;
-            }
         }
     }
-    return number;
+
```

Comment:
Changes to this script feel unrelated to Sonarcloud? (And have broken the tests :stuck_out_tongue: )

Reply:
This function has a "critical" bug, which was what this is trying to address. So it is related, but it needs more work clearly. Draft PR FTW. :P